packages/react/presence/src/Presence.tsx

```diff
@@ -90,18 +90,25 @@ function usePresence(present: boolean) {
        */
       const handleAnimationEnd = (event: AnimationEvent) => {
         const currentAnimationName = getAnimationName(stylesRef.current);
-        const isCurrentAnimation = event.animationName === currentAnimationName;
+        const isCurrentAnimation = currentAnimationName.includes(event.animationName);
         if (event.target === node && isCurrentAnimation) {
-          // if animation occurred, store its name as the previous animation
-          prevAnimationNameRef.current = getAnimationName(stylesRef.current);
           send('ANIMATION_END');
         }
       };
 
+      const handleAnimationStart = (event: AnimationEvent) => {
+        if (event.target === node) {
+          // if animation occurred, store its name as the previous animation.
+          prevAnimationNameRef.current = getAnimationName(stylesRef.current);
```

Comment:
I had to move this to `animationstart` because if the animation in doesn't complete before the button is pressed again to close, the `animationend` wouldn't have fired so the `prevAnimationName` would be incorrect.

Reply:
Just to confirm, this change is unrelated to having to support multiple animations right? Just an edge case you discovered whilst fixing this, in case you toggle the state quickly?

Reply:
You got it :+1: